easybuild/easyconfigs/t/Theano/Theano-1.1.2-foss-2021b-PyMC.eb

```diff
@@ -0,0 +1,40 @@
+easyblock = 'PythonPackage'
+
+name = 'Theano'
+version = '1.1.2'
+versionsuffix = '-PyMC'
+
+homepage = 'https://deeplearning.net/software/theano'
+description = """Theano is a Python library that allows you to define, optimize,
+and evaluate mathematical expressions involving multi-dimensional arrays efficiently."""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+
+source_urls = ['https://pypi.python.org/packages/source/T/Theano%(versionsuffix)s']
+sources = ['Theano-PyMC-%(version)s.tar.gz']
+checksums = ['5da6c2242ea72a991c8446d7fe7d35189ea346ef7d024c890397011114bf10fc']
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+]
+
+download_dep_fail = True
+use_pip = True
+sanity_pip_check = True
+
+fix_python_shebang_for = ['bin/theano-cache', 'bin/theano-nose']
+
+sanity_check_paths = {
+    'files': ['bin/theano-cache'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+local_blas_warning = "Using NumPy C-API based implementation for BLAS functions"
+sanity_check_commands = [
+    # make sure that warning about falling back to (slow) NumPy C-API for BLAS functions is *not* printed
+    # mkl-service dependency is required to ensure this doesn't happen
+    "python -c 'import theano.tensor' 2>&1 | grep '%s'; test $? != 0" % local_blas_warning,
```

Comment:
Hmm, seems like this passes even though `mkl-service` is not included as a dependency (anymore)...
We should check this a bit more carefully.

Does this check in `Theano-1.1.2-intel-2020b-PyMC.eb` fail if `mkl-service` is not used as a dependency?

Please test with

```
eb Theano-1.1.2-intel-2020b-PyMC.eb --filter-deps mkl-serivce --rebuild
```

Reply:
OK, it seems I was mixing things up a bit here...

`mkl-service` is only need when installing Theano with the `intel` toolchain; if `mkl-service` is removed, this sanity check command fails, because a fallback to the (very slow) NumPy C-API for BLAS fuctions is detected.

When using the `foss` toolchain to install Theano however, `mkl-service` is irrelevant because `MKL` is not a part of the `foss` toolchain. And hence this sanity check passes even when not including `mkl-service` as a dependency. There is no fallback to the slow NumPy C-API for BLAS in this case, since the OpenBLAS component in `foss` is used (and that works fine without the `mkl-service` "trick").

So, long story short: this PR is probably ready to go pretty much, I'll give it another review/test, but this particular comment is not a blocker for this PR at all...